lib/utils.js

```diff
@@ -67,15 +67,46 @@ exports.GreatError.prototype.toJSON = function () {
 
 
 // Payload for "response" events
-exports.GreatResponse = function (request, options) {
+exports.GreatResponse = function (request, options, filterRules) {
 
     if (this.constructor !== exports.GreatResponse) {
-        return new exports.GreatResponse(request, options);
+        return new exports.GreatResponse(request, options, filterRules);
     }
 
     var req = request.raw.req;
     var res = request.raw.res;
 
+    var replacer = function (match, group1) {
+
+        return (new Array(group1.length + 1).join('X'));
+    };
+
+    var applyFilter = function (data) {
+
+        for (var key in data) {
+            if (typeof data[key] === 'object') {
+                return applyFilter(data[key]);
+            }
+
+            // there is a filer for this key, so we are going to update the data
+            if (filterRules[key]) {
+                var filter = filterRules[key].toLowerCase();
+
+                if (filter === 'censor') {
+                    data[key] = ('' + data[key]).replace(/./gi, 'X');
```

Comment:
Don't need the `i` since `.` since you don't need to worry about case-insensitive